src/vcastUtilities.ts

```diff
@@ -159,49 +162,71 @@ export function addIncludePath (fileUri: vscode.Uri) {
   // I'm handling a few error cases here without going crazy
   let statusMessages:string[] = [];
 
-  // read the existing file contents
   let existingJSON: any;
-  try {
-    existingJSON = JSON.parse(fs.readFileSync(fileUri.fsPath));
-    if (existingJSON.configurations.length == 0) {
-      statusMessages.push (`{configurationFile} file has no existing configurations, creating a 'vcast' configuraiton.  `); 
-      existingJSON.configurations.push ({name: "vcast", includePath: []});
-    }
-  } catch {
-    // if there is some sort of parse error with the existing file don't change it
-    vscode.window.showErrorMessage(`Exception parsing {configurationFile} file, no changes made.  Check for syntax errors.  `);
+  let existingJSONasString: string;
+  
+  // Requires json-c parsing to handle comments etc.
+  existingJSONasString  = fs.readFileSync (fileUri.fsPath).toString();
+  var parseErrors: jsonc.ParseError[] = [];  // not using programatically, for debug only
+  // note that jsonc.parse returns "real json" without the comments
+  existingJSON = jsonc.parse(existingJSONasString, parseErrors, jsoncParseOptions);
+
+  if (existingJSON && existingJSON.configurations && existingJSON.configurations.length > 0) {
```

Comment:
Okay, so there _is_ a chance we don't have `configurations`?

Can we please review this code to check the consistency of checking  `configurations`? We probably want `existingJSON && existingJSON.configurations && existingJSON.configurations.length > 0` everywhere (or a helper that encapsulates this)?